### ISSUE TITLE
Rename "Value expressions and place expressions" section to "Evaluation categories"

### ISSUE THREAD

**Issue by Havvy** [rust-lang/reference#972]
C++ calls them [value categories](https://docs.microsoft.com/en-us/cpp/cpp/lvalues-and-rvalues-visual-cpp?view=msvc-160). Since we call one "value expression", I'd like to avoid that term and instead call them a category of some kind. Since the distinction is about what kind of thing the expression evaluates to, I figured "evaluation category" would be the best term.

Having this term would also let me say parenthesized expressions are of the same evaluation category as their enclosed operand.

**Comment by Havvy** [comment]
Nominating with the question that adding this term is correct and that the name is alright.

**Comment by nikomatsakis** [comment]
Perhaps "categories of expressions" or "expression categories"?

**Comment by nikomatsakis** [comment]
Evaluation seems strange to me because it sounds very much like a property of what happens when the expression *executes*, but this is more of a category of expression that *can* be evaluated to a place in memory in some contexts.

**Comment by joshtriplett** [comment]
Agreed. We're talking about a name that encompasses "value expression" and "place expression", so "expression category" seems reasonable.
